Objects/unicodeobject.c

```diff
@@ -2458,13 +2458,15 @@ unicode_adjust_maxchar(PyObject **p_unicode)
         if (max_char >= 256)
             return;
     }
-    else {
+    else if (kind == PyUnicode_4BYTE_KIND) {
```

Comment:
Maybe a switch/case would be more appropriate.

Reply:
It was just a remark. You can leave the code as it is, I already approved your PR.